### PR TITLE
Add a button to Qt Error Notification to show tracebacks.

### PR DESCRIPTION
**Pull request by Carreau**
And have the "View Traceback", button localized.

# Description

Make it easier to see traceback when using the GUI. 
Notification button now have a "View traceback", that will pop up a window showing the traceback

## Type of change

- [ ] Bug-fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How has this been tested?

- [ ] example: the test suite for my feature covers cases x, y, and z
- [ ] example: all tests pass with my change

## Final checklist:
- [x] My PR is the minimum possible work for the desired functionality
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Comment by goanpeca** [comment]
Awesome @Carreau, could you add a screenshot for the before and after :) ?

**Comment by Carreau** [comment]
Screen Shot 2021-03-22 at 12 51 48 PM
Adding the button expand the notification by default.
Screen Shot 2021-03-22 at 12 52 27 PM

Here is the pop-up window once you click on "View Traceback"; 
Screen Shot 2021-03-22 at 12 52 20 PM

I still need to write some tests.

```
from napari._qt.widgets.qt_viewer_buttons import QtViewerPushButton
btn = QtViewerPushButton(
                viewer, 'warning', 'new Error', lambda: None
            )
def raise_():
    raise ValueError("error!")
btn.clicked.connect(raise_)
qtv = viewer.window.qt_viewer
l = qtv.layerButtons.children()[0]
l.addWidget(btn)
```

can be used to add a button that raise an error from the console.

**Comment by tlambert03** [comment]
> can be used to add a button that raise an error from the console.

by the way @Carreau, I added your error-button debug code to `examples/dev/gui_notifications.py` ... feel free to use/update that as necessary when you want to test this stuff

**Comment by Carreau** [comment]
> by the way @Carreau, I added your error-button debug code to `examples/dev/gui_notifications.py` ... feel free to use/update that as necessary when you want to test this stuff

Ah, great, that works well; 


I've added test that error have a "View Traceback" button; I'm not sure how to get to the dialog that is shown when we click on the button though, and if it is worth it to test it correctly show the exceptions.

File: napari/_qt/_tests/test_qt_notifications.py
Comment (tlambert03) [inline napari/_qt/_tests/test_qt_notifications.py]:
```suggestion
    bttn = dialog.row2_widget.findChild(QPushButton)
    assert bttn.text() == 'View Traceback'
    mock_show.assert_not_called()
    bttn.click()
    mock_show.assert_called_once()
```

Comment (tlambert03) [reply napari/_qt/_tests/test_qt_notifications.py]: (slight improvement... at least confirms that clicking the button shows a dialog...).  as I'm sure you found in this case, it's tough to grab the dialog here since without a viewer instance, there is no `parent.parent()`.  We can patch that to make it easier later if necessary... but I'm happy with this test.  thanks!

**Comment by tlambert03** [comment]
This is looking great @Carreau, thank you!  I did notice one slight issue (an existing issue likely, but something that this exacerbates), which is that the caret button that shows/hides the 2nd detail not working, and sizeHint isn't working. It _should_ hide the second row, and show only a single line of elided text, and then expand when clicking that caret... but currently it's showing too much, and the toggle has weird behavior:

https://user-images.githubusercontent.com/1609449/112063893-7d353300-8b38-11eb-80b3-0c481e6394fc.mov

on master (though there's nothing useful about the button), it's not doing that:


https://user-images.githubusercontent.com/1609449/112063922-8cb47c00-8b38-11eb-98e0-bdc77d8a28f7.mov

I'll try to have a look at this later if you don't figure it out in the meantime

**Comment by tlambert03** [comment]
on further investigation, I _think_ (but still not sure) that this is an error in the `sizeHint` of the elided label.  interestingly, it seems to work fine if the error message is longer than one line.  Anyway, I'll try to fix this in another PR.  don't worry about it here.  we can live with the weird caret behavior for a little bit

File: napari/_qt/_tests/test_qt_notifications.py
Comment (Carreau) [reply napari/_qt/_tests/test_qt_notifications.py]: ah, findChild is useful thanks ! I was looking for something like this.

**Comment by Carreau** [comment]
Yeah, I saw this issue too with the hiding; but figured it was not related to adding the button. 
If you do not get to it I can have a look as well.

**Review by sofroniewn — APPROVED** [review]
Looks great! Very exciting. I'm approving now, but will wait till tomorrow morning to merge in case there are any further comments!!

**Review by tlambert03 — APPROVED** [review]

**Pull request merged**
